### PR TITLE
[ISSUE #3112] add switch configuration for plugins to enable it or not

### PR DESCRIPTION
**Pull request by hgaol**
// Describe your PR here; eg. Fixes #issueNo

For #3112 , added below configurations. e.g. If you set `shenyu.context-path.enabled=false`, it will be disabled. Otherwise, it will be enabled.

| plugin                | configuration                           | default |
| --------------------- | --------------------------------------- | ------- |
| ContextPathPlugin     | shenyu.plugins.context-path.enabled     | true    |
| CryptorRequestPlugin  | shenyu.plugins.cryptor.enabled          | true    |
| DividePlugin          | shenyu.plugins.divide.enabled           | true    |
| AlibabaDubboPlugin    | shenyu.plugins.alibaba-dubbo.enabled    | true    |
| ApacheDubboPlugin     | shenyu.plugins.apache-dubbo.enabled     | true    |
| GeneralContextPlugin  | shenyu.plugins.general-context.enabled  | true    |
| GrpcPlugin            | shenyu.plugins.grpc.enabled             | true    |
| HystrixPlugin         | shenyu.plugins.hystrix.enabled          | true    |
| JwtPlugin             | shenyu.plugins.jwt.enabled              | true    |
| LoggingPlugin         | shenyu.plugins.logging.enabled          | true    |
| LoggingRocketMQPlugin | shenyu.plugins.logging.rocketmq.enabled | true    |
| ModifyResponsePlugin  | shenyu.plugins.modify-response.enabled  | true    |
| MotanPlugin           | shenyu.plugins.motan.enabled            | true    |
| MqttPlugin            | shenyu.plugins.mqtt.enabled             | true    |
| OAuth2Plugin          | shenyu.plugins.oauth2.enabled           | true    |
| ParamMappingPlugin    | shenyu.plugins.param-mapping.enabled    | true    |
| RateLimiterPlugin     | shenyu.plugins.rete-limiter.enabled     | true    |
| RedirectPlugin        | shenyu.plugins.redirect.enabled         | true    |
| RequestPlugin         | shenyu.plugins.request.enabled          | true    |
| Resilience4JPlugin    | shenyu.plugins.resilience4j.enabled     | true    |
| ResponsePlugin        | shenyu.plugins.response.enabled         | true    |
| RewritePlugin         | shenyu.plugins.rewrite.enabled          | true    |
| SentinelPlugin        | shenyu.plugins.sentinel.enabled         | true    |
| SignPlugin            | shenyu.plugins.sign.enabled             | true    |
| SofaPlugin            | shenyu.plugins.sofa.enabled             | true    |
| SpringCloudPlugin     | shenyu.plugins.spring-cloud.enabled     | true    |
| TarsPlugin            | shenyu.plugins.tars.enabled             | true    |
| URIPlugin             | shenyu.plugins.uri.enabled              | true    |
| WafPlugin             | shenyu.plugins.waf.enabled              | true    |
| WebSocketPlugin       | shenyu.plugins.web-socket.enabled       | true    |

Make sure that:

- [x] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor).
- [ ] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed `mvn clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: shenyu-spring-boot-starter/shenyu-spring-boot-starter-plugin/shenyu-spring-boot-starter-plugin-context-path/src/main/java/org/apache/shenyu/springboot/starter/plugin/context/path/ContextPathPluginConfiguration.java [hunk 1]
```diff
@@ -21,13 +21,15 @@
...
 import org.apache.shenyu.plugin.context.path.handler.ContextPathPluginDataHandler;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 
 /**
  * The type Context path plugin configuration.
  */
 @Configuration
+@ConditionalOnProperty(value = {"shenyu.contextPath.enabled"}, havingValue = "true", matchIfMissing = true)
```
Comment (yu199195) [inline shenyu-spring-boot-starter/shenyu-spring-boot-starter-plugin/shenyu-spring-boot-starter-plugin-context-path/src/main/java/org/apache/shenyu/springboot/starter/plugin/context/path/ContextPathPluginConfiguration.java:32]: all plugin add  shenyu.plugins.xxxxx.enabled

Comment (hgaol) [reply shenyu-spring-boot-starter/shenyu-spring-boot-starter-plugin/shenyu-spring-boot-starter-plugin-context-path/src/main/java/org/apache/shenyu/springboot/starter/plugin/context/path/ContextPathPluginConfiguration.java:32]: Got it. Should I also change the existing configurations? Like `shenyu.metrics.enabled` to `shenyu.plugins.metrics.enabled`, as well as other configurations under `shenyu.metrics`. `shenyu.logging.rocketmq` is also such scenario.

Comment (hgaol) [reply shenyu-spring-boot-starter/shenyu-spring-boot-starter-plugin/shenyu-spring-boot-starter-plugin-context-path/src/main/java/org/apache/shenyu/springboot/starter/plugin/context/path/ContextPathPluginConfiguration.java:32]: BTW, `shenyu.plugins` or `shenyu.plugin`? Just saw you mentioned `shenyu.plugin` in mail.

Comment (yu199195) [reply shenyu-spring-boot-starter/shenyu-spring-boot-starter-plugin/shenyu-spring-boot-starter-plugin-context-path/src/main/java/org/apache/shenyu/springboot/starter/plugin/context/path/ContextPathPluginConfiguration.java:32]: 1. shenyu.metrics.enable ，can not change
2. shenyu.plugins.logging-rocketmq.enabled

**Comment by midnight2104** [comment]
pls check ci   https://github.com/apache/incubator-shenyu/runs/5679421019?check_suite_focus=true

**Comment by hgaol** [comment]
> pls check ci https://github.com/apache/incubator-shenyu/runs/5679421019?check_suite_focus=true

done.

File: shenyu-spring-boot-starter/shenyu-spring-boot-starter-plugin/shenyu-spring-boot-starter-plugin-websocket/src/main/java/org/apache/shenyu/springboot/plugin/websocket/WebSocketPluginConfiguration.java [hunk 2]
```diff
@@ -35,7 +35,7 @@
  * The type Web socket plugin configuration.
  */
 @Configuration
-@ConditionalOnProperty(value = {"shenyu.webSocket.enabled"}, havingValue = "true", matchIfMissing = true)
+@ConditionalOnProperty(value = {"shenyu.plugins.web-socket.enabled"}, havingValue = "true", matchIfMissing = true)
```
Comment (yu199195) [inline shenyu-spring-boot-starter/shenyu-spring-boot-starter-plugin/shenyu-spring-boot-starter-plugin-websocket/src/main/java/org/apache/shenyu/springboot/plugin/websocket/WebSocketPluginConfiguration.java:38]: websocket

Comment (hgaol) [reply shenyu-spring-boot-starter/shenyu-spring-boot-starter-plugin/shenyu-spring-boot-starter-plugin-websocket/src/main/java/org/apache/shenyu/springboot/plugin/websocket/WebSocketPluginConfiguration.java]: my mistake, resolved.

**Review by yu199195 — APPROVED** [review]

**Review by yu199195 — APPROVED** [review]

**Pull request merged**